elasticdl/python/tests/feature_column_test.py

```diff
@@ -36,13 +36,52 @@ def test_feature_column_call(self):
         output = call_feature_columns(
             [item_id_embedding], {"item_id": [1, 2, 3]}
         )
+
         self.assertTrue(
             (
                 output.numpy()
                 == generate_mock_embedding_vectors([1, 2, 3], dimension)
             ).all()
         )
 
+    def test_call_embedding_column_with_weights(self):
+        dimension = 8
+
+        item_id_embedding = feature_column.embedding_column(
+            tf.feature_column.weighted_categorical_column(
+                tf.feature_column.categorical_column_with_identity(
+                    "item_id", num_buckets=128
+                ),
+                weight_feature_key="frequency",
+            ),
+            dimension=dimension,
+            initializer=tf.initializers.identity,
+            combiner="sum",
+        )
+        item_id_embedding.lookup_embedding = lambda unique_ids: (
+            generate_mock_embedding_vectors(
+                unique_ids, item_id_embedding.dimension
+            )
+        )
+
+        output = call_feature_columns(
+            [item_id_embedding],
+            {
+                "item_id": [[2, 6, 5], [3, 1, 1]],
+                "frequency": [[0.33, 5.0, 1.024], [2.048, 0.5, 1.0]],
+            },
+        )
+
+        expected_output = np.array(
+            [
+                [0.0, 0.0, 0.33, 0.0, 0.0, 1.024, 5.0, 0.0],
+                [0.0, 1.5, 0.0, 2.048, 0.0, 0.0, 0.0, 0.0],
+            ],
+            dtype=np.float32,
+        )
+
+        self.assertTrue((output.numpy() == expected_output).all())
```

Comment:
You can try `numpy.array_equal`. https://docs.scipy.org/doc/numpy/reference/generated/numpy.array_equal.html

Reply:
Good idea, will have a try.